src/engineChange.js

```diff
@@ -57,15 +57,24 @@ function loadStoredEngine() {
     var engine = EngineVersions[engineId];
     var queryVariable = getQueryVariable("engine_" + engineId);
 
-    if (queryVariable && engine.versions[queryVariable] !== undefined) {
-      window.localStorage["engine_" + engineId] = queryVariable;
-      window.location.href = window.location.search.replace("engine_" + engineId + "=" + queryVariable, "");
-    } else if (window.localStorage["engine_" + engineId] === undefined) {
+    // if engine version is missing, add the default value from dict 'EngineVersions' to the url:
+    if (!queryVariable) {
       window.localStorage["engine_" + engineId] = engine.defaultVersion;
+      window.location.href += "&engine_" + engineId + "=" + engine.defaultVersion;
+      // if engine version is provided and it is the same in Local Storage do nothing:
+    } else if (queryVariable && window.localStorage["engine_" + engineId] === queryVariable) {
+      engine.defaultVersion = queryVariable;
     } else {
-      var version = window.localStorage["engine_" + engineId];
-      if (engine.versions[version] !== undefined) {
-        engine.defaultVersion = window.localStorage["engine_" + engineId];
+      // if engine version is provided but it does not exist in 'EngineVersions' dict, replace it in url to the default value:
```

Comment:
Please print a console warning/error in such case like:

Shaka version x.x.x is not supported, overriding with x.x.x

Reply:
Console warning added.